lib/src/state/dna_design.dart

```diff
@@ -553,7 +553,22 @@ abstract class DNADesign with UnusedFields implements Built<DNADesign, DNADesign
 
   bool has_nondefault_max_offset(Helix helix) {
     var ends = domains_on_helix(helix.idx).map((ss) => ss.end);
-    int max_end = ends.isEmpty ? constants.default_max_offset : ends.reduce(max);
+    int max_end = 0;
+    int greatest_end_val = 0;
+    for (var strand in strands){
+          for(var domain in strand.domains()){
+            if((domain.end > greatest_end_val)){
+              greatest_end_val = domain.end;
+            }
+          }  
+        }
+    if(ends.isEmpty){
```

Comment:
I'm still not sure this is correct.

`ends` is the ends on just this one helix. It could be empty, but if there are ends on *other* helices, then `max_offset` should not be `constants.default_max_offset`. That constant is only used if there are no strands in the whole system.

We'll take this as an opportunity to do [test-driven development](https://en.wikipedia.org/wiki/Test-driven_development).

*Before* fixing the code, *first* write tests to produce this error. If the unit test code makes sense to you, can you first write a test to capture this? In other words, make up a design that has three helices, one with no ends, one with max end offset 10, and another with max end offset 20. The test should verify that all three helices get assigned `max_offset` equal to 20. If the test is written correctly, it will fail when it tests the empty helix in that design.

Then, once the unit test reproduces the error, fix the error in this line of code.

We can meet and I can explain how to do that. I recommend testing only the `max_offset`'s, instead of the style in the current tests, where the entire `AppState` is tested for equality with the expected state.